Week 3/Solution.java

```diff
@@ -0,0 +1,64 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+    public List<List<Integer>> threeSum(int[] n) {
+        List<List<Integer>> res = new  ArrayList<List<Integer>>();
+        if(n == null || n.length<3) return res;
+        int len = n.length;
+        HashSet<Integer> checked = new HashSet<Integer>();
+        for(int i = 0; i<len; i++){
+            int goal = n[i]*-1;
```

Comment:
this was how I ended up solving it too, yay! one suggestion I have is putting some more comments in your code. also, good idea to use `HashSet<Int>` to keep track of duplicates in the input array. 👍